### PR TITLE
Expire listings after 30 days

### PR DESCRIPTION
**Pull request by fatlotus**
This one ended up being trickier than I had expected :(

Main changes:
1. Listings are no longer bumped to the top when edited.
2. Listings are still visible, but do not come up in searches, after 30 days.

#209 #119 #216

### CODE REVIEW

**Comment by georgeteo** [comment]
1. Do we want a way of pushing listings to the top? ("Refresh listing")?

File: caravel/daemons/migration.py [hunk 1]
```diff
@@ -16,8 +17,12 @@ def grouper(iterable, n, fillvalue=None):
 
 @app.route("/_internal/migrate_schema")
 def migrate_schema():
+    horizon = datetime.datetime.now() - model.Listing.MARK_AS_OLD_AFTER
```
Comment (georgeteo) [inline caravel/daemons/migration.py]: This migration will cause a lot of things to disappear. Would it be useful to send an email to the seller or post an alert on the landing page?

Comment (fatlotus) [inline caravel/daemons/migration.py]: Hm... maybe. Alternatively, we could reduce it to only index by seller.

**Comment by fatlotus** [comment]
1. Probably :( 'Nother pull request?
2. Yes. This is the main goal, actually :p

**Comment by fatlotus** [comment]
aaand there's a weird bug

https://caravel-code-reviews.appspot.com/92f96446-1a0f-415c-bdcf-6de60edc1c25

**Comment by fatlotus** [comment]
Weird bug is now fixed.

**Comment by fatlotus** [comment]
Weeee! Bump button added. #208

**Pull request merged**